### PR TITLE
Release v0.5.0 version as stable

### PR DESCRIPTION
**Pull request by davidz627**
newest release version been cut, update deployment specs and docs.

/assign @msau42 @verult

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,9 +18,6 @@
 STAGINGIMAGE=${GCE_PD_CSI_STAGING_IMAGE}
 STAGINGVERSION=${GCE_PD_CSI_STAGING_VERSION}
 
-PRODIMAGE=gcr.io/gke-release/gcp-compute-persistent-disk-csi-driver
```
Comment (verult) [inline Makefile]: Why were these lines removed?

Comment (davidz627) [reply Makefile]: we aren't building a prod image from the makefile ever (we use the internal release flows). I wanted to keep them around for "documentation" purposes for a while, but now our actual documentation captures this same information so I think its reasonable to remove

**Comment by davidz627** [comment]
/test pull-gcp-compute-persistent-disk-csi-driver-e2e
cmon prow you can do this
